projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/datajobs/DeploymentModelConverter.java

```diff
@@ -101,6 +105,15 @@ public static JobDeployment mergeDeployments(
         newDeployment.getVdkVersion() != null
             ? newDeployment.getVdkVersion()
             : oldDeployment.getVdkVersion());
+
+    if (oldDeployment.getPythonVersion() == null && newDeployment.getPythonVersion() == null) {
+      mergedDeployment.setPythonVersion(new SupportedPythonVersions().getDefaultPythonVersion());
```

Comment:
```suggestion
      mergedDeployment.setPythonVersion(new supportedPythonVersions.getDefaultPythonVersion());
```

Reply:
Hi, @tozka

This will not work, as the context is static, but the methods in the SupportedPythonVersions class are not.

Reply:
The `new SupportedPythonVersions()` approach won't work as well since the Spring context won't be initialized and the default version will be null. This is not the right place to enforce the default version which is business logic and should be moved to the DeploymentService.

Reply:
+1 what Miro said. The method purpose is pretty clear "Merge two deployments with preference of fields from newDeployment over oldDeployment" 

Setting the default version should be done in another place.

Reply:
Moved the logic to the service. Thanks for the reviews